src/lib/ThumbnailsSidebar.js

```diff
@@ -289,19 +297,21 @@ class ThumbnailsSidebar {
      * @return {void}
      */
     requestThumbnailImage(itemIndex, thumbnailEl) {
-        requestAnimationFrame(() => {
-            this.createThumbnailImage(itemIndex).then(imageEl => {
-                // Promise will resolve with null if create image request was already in progress
-                if (imageEl) {
-                    // Appends to the thumbnail nav element
-                    thumbnailEl.lastChild.appendChild(imageEl);
-                    thumbnailEl.classList.add(CLASS_BOX_PREVIEW_THUMBNAIL_IMAGE_LOADED);
-                }
-
-                // After generating the thumbnail image, render the next one
-                this.renderNextThumbnailImage();
-            });
-        });
+        this.animationFrameRequestIds.push(
+            requestAnimationFrame(() => {
+                this.createThumbnailImage(itemIndex).then(imageEl => {
```

Comment:
We'll need to remove the requestId from `animationFrameRequestIds` before this line as well to avoid an ever-growing array of already-resolved ids.